### PR TITLE
docs: adapt README according to general template

### LINKED ISSUE #119: Update README to be consistent with other SDKs
Update readme to be consistent with the template here: https://github.com/open-feature/.github/pull/36

DoD:

- README structure matches tempalte
- Examples are up-to-date and added for all `TODOs` in template
- details only pertaining to developers of the SDK are moved to CONTRIBUTING.md

See similar Java PR: https://github.com/open-feature/java-sdk/pull/382

### PR DESCRIPTION
**Pull request by odubajDT**
## This PR

- adapts readme according to general template
- adds missing sections

### Related Issues

Fixes #119

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,75 @@
...
+<h2 align="center">OpenFeature .NET SDK</h2>
 
 [![a](https://img.shields.io/badge/slack-%40cncf%2Fopenfeature-brightgreen?style=flat&logo=slack)](https://cloud-native.slack.com/archives/C0344AANLA1)
 [![spec version badge](https://img.shields.io/badge/Specification-v0.5.2-yellow)](https://github.com/open-feature/spec/tree/v0.5.2?rgh-link-date=2023-01-20T21%3A37%3A52Z)
 [![codecov](https://codecov.io/gh/open-feature/dotnet-sdk/branch/main/graph/badge.svg?token=MONAVJBXUJ)](https://codecov.io/gh/open-feature/dotnet-sdk)
 [![nuget](https://img.shields.io/nuget/vpre/OpenFeature)](https://www.nuget.org/packages/OpenFeature)
 [![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/6250/badge)](https://bestpractices.coreinfrastructure.org/projects/6250)
 
-OpenFeature is an open standard for feature flag management, created to support a robust feature flag ecosystem using cloud native technologies. OpenFeature will provide a unified API and SDK, and a developer-first, cloud-native implementation, with extensibility for open source and commercial offerings.
+## 👋 Hey there! Thanks for checking out the OpenFeature Java SDK
```
Comment (bacherfl) [inline README.md:18]:
```suggestion
## 👋 Hey there! Thanks for checking out the OpenFeature .NET SDK

```

File: README.md [hunk 2]
```diff
@@ -1,20 +1,75 @@
...
-## Supported .Net Versions
+## 🔍 Requirements:
 
-The packages will aim to support all current .NET versions. Refer to the currently supported versions [.NET](https://dotnet.microsoft.com/download/dotnet) and [.NET Framework](https://dotnet.microsoft.com/download/dotnet-framework) Excluding .NET Framework 3.5
+- .NET 6+
+- .NET Core 6+
+- .NET Framework 4+
 
-## Getting Started
+Note that he packages will aim to support all current .NET versions. Refer to the currently supported versions [.NET](https://dotnet.microsoft.com/download/dotnet) and [.NET Framework](https://dotnet.microsoft.com/download/dotnet-framework) Excluding .NET Framework 3.5
```
Comment (bacherfl) [inline README.md:34]:
```suggestion
Note that the packages will aim to support all current .NET versions. Refer to the currently supported versions [.NET](https://dotnet.microsoft.com/download/dotnet) and [.NET Framework](https://dotnet.microsoft.com/download/dotnet-framework) excluding .NET Framework 3.5

```

**Review by bacherfl — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,20 +1,76 @@
...
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+</Project>
+```
+<!-- x-release-please-end-version -->
+
+### Software Bill of Materials (SBOM)
+
+We publish SBOMs with all of our releases as of 0.3.0. You can find them in Maven Central alongside the artifacts.
```
Comment (kinyoklion) [inline README.md:62]: Is this a copy paste mistake from java?

**Review by kinyoklion — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -1,20 +1,72 @@
...
 
-## Supported .Net Versions
+## 📦 Installation:
 
-The packages will aim to support all current .NET versions. Refer to the currently supported versions [.NET](https://dotnet.microsoft.com/download/dotnet) and [.NET Framework](https://dotnet.microsoft.com/download/dotnet-framework) Excluding .NET Framework 3.5
+Use the following in your `*.csproj` file:
 
-## Getting Started
+<!-- x-release-please-start-version -->
+```xml
```
Comment (toddbaert) [inline README.md:42]: Can you update this to include some commands, and less XML? 

See example here from contribs: https://github.com/open-feature/dotnet-sdk-contrib/tree/main/src/OpenFeature.Contrib.Providers.Flagd

Comment (odubajDT) [reply README.md]: fixed, thx!

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**